packages/api/tests/serial/endpoints/test-granules.js

```diff
@@ -680,33 +703,126 @@ test.serial('move a file and update metadata', async (t) => {
+      filename: `s3://${publicBucket}/${process.env.stackName}/original_filepath/${newGranule.granuleId}.cmr.json`
+    }
+  ];
+
+  await granuleModel.create(newGranule);
+
+  await Promise.all(newGranule.files.map((file) => {
+    if (file.name === `${newGranule.granuleId}.txt`) {
+      return putObject({ Bucket: file.bucket, Key: file.filepath, Body: 'test data' });
+    }
+    return putObject({ Bucket: file.bucket, Key: file.filepath, Body: ummgMetadataString });
+  }));
+
+  const destinationFilepath = `${process.env.stackName}/moved_granules`;
+  const destinations = [
+    {
+      regex: '.*.txt$',
+      bucket: internalBucket,
+      filepath: destinationFilepath
+    }
+  ];
+
+  sinon.stub(
+    CMR.prototype,
+    'ingestUMMGranule'
+  ).returns({ result: { 'concept-id': 'id204842' } });
+
+  const response = await request(app)
+    .put(`/granules/${newGranule.granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${accessToken}`)
+    .send({
+      action: 'move',
+      destinations
+    })
+    .expect(200);
+
+  const body = response.body;
+
+  t.is(body.status, 'SUCCESS');
+  t.is(body.action, 'move');
+
+  // text file has moved to correct location
+  const list = await aws.s3().listObjects({
+    Bucket: internalBucket,
+    Prefix: destinationFilepath
   }).promise();
-  await aws.recursivelyDeleteS3Bucket(buckets.public.name);
-  return new Promise((resolve, reject) => {
-    xml2js.parseString(file.Body, xmlParseOptions, (err, data) => {
-      if (err) return reject(err);
-      return resolve(data);
-    });
-  }).then((xml) => {
-    const newUrls = xml.Granule.OnlineAccessURLs.OnlineAccessURL.map((obj) => obj.URL);
-    const newDestination = `${process.env.DISTRIBUTION_ENDPOINT}${destinations[0].bucket}/${destinations[0].filepath}/${newGranule.files[0].name}`;
-    t.true(newUrls.includes(newDestination));
+  t.is(list.Contents.length, 1);
+  t.is(list.Contents[0].Key.indexOf(destinationFilepath), 0);
+
+  // CMR JSON  is in same location.
+  const list2 = await aws.s3().listObjects({
+    Bucket: publicBucket,
+    Prefix: `${process.env.stackName}/original_filepath`
+  }).promise();
+  t.is(list2.Contents.length, 1);
+  t.is(newGranule.files[1].filepath, list2.Contents[0].Key);
+
+  // CMR UMMG JSON has been updated with the location of the moved file.
+  const ummgObject = await metadataObjectFromCMRJSONFile(newGranule.files[1].filename);
+  const updatedURLs = ummgObject.RelatedUrls.map((urlObj) => urlObj.URL);
+  const newDestination = `${process.env.DISTRIBUTION_ENDPOINT}${destinations[0].bucket}/${destinations[0].filepath}/${newGranule.files[0].name}`;
+  t.true(updatedURLs.includes(newDestination));
+
+  // Original metadata is also unchanged.
+  const origURLs = originalUMMG.RelatedUrls.map((urlObj) => urlObj.URL);
+  origURLs.forEach((origURL) => {
+    t.true(updatedURLs.includes(origURL));
```

Comment:
What's the point of this test?

Reply:
No, this is actually testing that the metadata that existed in the initial ummg.json file was not deleted when we modified it by adding/updating the information from the moved files.

In the initial code (from long ago) you can see that all of the OnlineAccessURLs were deleted and only updated/moved urls were then added. 
https://github.com/nasa/cumulus/blob/v1.10.1/packages/ingest/granule.js#L744

It's possible that the initial data has OnlineAccessURLs that aren't directly related to the files moved.
e.g.

```
 {
   "URLContentType": "CollectionURL",
   "Type": "DATA SET LANDING PAGE",
   "URL": "http://dx.doi.org/10.5067/HS3/HIRAD/DATA201"
 },
 {
   "URLContentType": "PublicationURL",
   "Type": "VIEW RELATED INFORMATION",
   "Subtype": "DATA CITATION POLICY",
   "URL": "https://ghrc.nsstc.nasa.gov/home/about-ghrc/citing-ghrc-daac-data"
 }
```

So this test, just reads all of the relatedUrls from the sample input ummg file , and makes sure they are still there when after we have written the new/updated Urls to the file.